<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from kuka_lwr_arm.urdf.xacro        | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="kuka">
    <!-- Remove this stuff when we get out of quasi-static -->
    <link name="world"/>

    <joint name="initial" type="fixed">
      <parent   link="world"/>
      <child    link="base_link"/>
      <origin   xyz="0 0 0" rpy="0 0 0"/>
    </joint>

  <!-- right is either 1 (for right arm) or -1 (for left arm) -->
  <link name="base_link">
    <inertial>
      <mass value="0"/>
      <!-- static base, disable dynamics for this link -->
      <origin rpy="0 0 0" xyz="0 0 0.055"/>
      <inertia ixx="0.00381666666667" ixy="0" ixz="0" iyy="0.0036" iyz="0" izz="0.00381666666667"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_base.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/base_link.dae"/>
      </geometry>
      <!-- <material name="Orange"/> -->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_base_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/base_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_a1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.11"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="1.91986217719"/>
    <dynamics damping="0.1"/>
    <parent link="base_link"/>
    <child link="link_1"/>
  </joint>
  <link name="link_1">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 0.04 0.130"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.0118666666667" iyz="0" izz="0.003"/>
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_segment_a.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/link_1.dae"/>
      </geometry>
      <!-- <material name="Orange"/> -->
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_segment_a_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/link_1.stl"/>
      </geometry>
    </collision>
  </link>
  <transmission name="trans_0" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="motor_0"/>
    <joint name="joint_a1"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="joint_a2" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20"/>
    <axis xyz="0 -1 0"/>
    <dynamics damping="0.1"/>
    <limit effort="0" lower="-2.0942" upper="2.0942" velocity="1.4834"/>
    <parent link="link_1"/>
    <child link="link_2"/>
  </joint>
  <link name="link_2">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 -0.04 0.07"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.0118666666667" iyz="0" izz="0.003"/>
    </inertial>
    <visual>
      <origin rpy="3.14159265359 0 3.14159265359" xyz="0 0 0.2"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_segment_b.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/link_2.dae"/>
      </geometry>
      <!-- <material name="Orange"/> -->
    </visual>
    <collision>
      <origin rpy="3.14159265359 0 3.14159265359" xyz="0 0 0.2"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_segment_b_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/link_2.stl"/>
      </geometry>
    </collision>
  </link>
  <transmission name="trans_1" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="motor_1"/>
    <joint name="joint_a2"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="joint_a3" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="2.26892802759"/>
    <dynamics damping="0.1"/>
    <parent link="link_2"/>
    <child link="link_3"/>
  </joint>
  <link name="link_3">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 -0.04 0.130"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.0118666666667" iyz="0" izz="0.003"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_segment_a.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/link_3.dae"/>
      </geometry>
      <!-- <material name="Orange"/> -->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_segment_a_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/link_3.stl"/>
      </geometry>
    </collision>
  </link>
  <transmission name="trans_2" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="motor_2"/>
    <joint name="joint_a3"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="joint_a4" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20"/>
    <axis xyz="0 1 0"/>
    <limit effort="306" lower="-2.09439510239" upper="2.09439510239" velocity="2.26892802759"/>
    <dynamics damping="0.1"/>
    <parent link="link_3"/>
    <child link="link_4"/>
  </joint>
  <link name="link_4">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 0.04 0.07"/>
      <inertia ixx="0.0136666666667" ixy="0" ixz="0" iyy="0.0118666666667" iyz="0" izz="0.003"/>
    </inertial>
    <visual>
      <origin rpy="0 3.14159265359 3.14159265359" xyz="0 0 0.2"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_segment_b.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/link_4.dae"/>
      </geometry>
      <!-- <material name="Orange"/> -->
    </visual>
    <collision>
      <origin rpy="0 3.14159265359 3.14159265359" xyz="0 0 0.2"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_segment_b_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/link_4.stl"/>
      </geometry>
    </collision>
  </link>
  <transmission name="trans_3" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="motor_3"/>
    <joint name="joint_a4"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="joint_a5" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.20"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="2.26892802759"/>
    <dynamics damping="0.1"/>
    <parent link="link_4"/>
    <child link="link_5"/>
  </joint>
  <link name="link_5">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 0.02 0.124"/>
      <inertia ixx="0.0126506666667" ixy="0" ixz="0" iyy="0.0108506666667" iyz="0" izz="0.003"/>
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry name="kuka_arm_5_geom">
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_segment_last.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/link_5.dae"/>
      </geometry>
      <!-- <material name="Orange"/> -->
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_segment_last_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/link_5.stl"/>
      </geometry>
    </collision>
  </link>
  <transmission name="trans_4" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="motor_4"/>
    <joint name="joint_a5"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="joint_a6" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.19"/>
    <axis xyz="0 -1 0"/>
    <limit effort="306" lower="-2.09439510239" upper="2.09439510239" velocity="3.14159265359"/>
    <dynamics damping="0.1"/>
    <parent link="link_5"/>
    <child link="link_6"/>
  </joint>
  <link name="link_6">
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.00520833333333" ixy="0" ixz="0" iyy="0.00520833333333" iyz="0" izz="0.00520833333333"/>
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_wrist.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/link_6.dae"/>
      </geometry>
      <!-- <material name="Grey"/> -->
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_wrist_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/link_6.stl"/>
      </geometry>
    </collision>
  </link>
  <transmission name="trans_5" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="motor_5"/>
    <joint name="joint_a6"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
  <joint name="joint_a7" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.078"/>
    <axis xyz="0 0 1"/>
    <limit effort="204" lower="-2.96705972839" upper="2.96705972839" velocity="3.14159265359"/>
    <dynamics damping="0.1"/>
    <parent link="link_6"/>
    <child link="link_7"/>
  </joint>
  <link name="link_7">
    <inertial>
      <mass value="2.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.000833333333333" ixy="0" ixz="0" iyy="0.000833333333333" iyz="0" izz="0.000833333333333"/>
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/arm_flanche.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/visual/link_7.dae"/>
      </geometry>
      <!-- <material name="Grey"/> -->
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <!-- <mesh filename="package://arm_game/meshes/kuka/convex/arm_flanche_convex.stl"/> -->
        <mesh filename="package://arm_game/meshes/lbr_iiwa_14_r820/collision/link_7.stl"/>
      </geometry>
    </collision>
  </link>
  <transmission name="trans_6" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="motor_6"/>
    <joint name="joint_a7"/>
    <mechanicalReduction>1.0</mechanicalReduction>
  </transmission>
</robot>
